CONTRIBUTING.md

```diff
@@ -1,18 +1,83 @@
+* If you want to submit a change or a patch, please read the contents below on how to make changes. New contributions are accepted as [Pull Requests](https://github.com/jscad/OpenJSCAD.org/pulls/) via GithHub.
 
-* If you have a question about how to use OpenJSCAD.org, please start a conversation at the [OpenJSCAD.org User Group](https://plus.google.com/communities/114958480887231067224). Please read the [OpenJSCAD.org User Guide](https://github.com/Spiritdude/OpenJSCAD.org/wiki/User-Guide) beforehand.
+* We only accept bug reports and pull requests on **GitHub**.
 
-* If you have a change or new feature in mind, please start a converstation with the [Core Developers](https://plus.google.com/communities/114958480887231067224) and start contributing changes.
+* If you have a question about how to use OpenJSCAD, then please start a conversation at the [OpenJSCAD.org User Group](https://plus.google.com/communities/114958480887231067224). You might find the answer in the [OpenJSCAD.org User Guide](https://github.com/Spiritdude/OpenJSCAD.org/wiki/User-Guide).
 
-Thanks! :heart: :heart: :heart:
+* If you have a change or new feature in mind, please start a conversation with the [Core Developers](https://plus.google.com/communities/114958480887231067224) and start contributing changes.
 
-The OpenJSCAD.org Core Developers
+Thanks!
+
+The JSCAD Organization
+
+## Making Changes
+
+First, we suggest that you fork this GIT repository. This will keep your changes separate from the fast lane.  And make pull requests easier.
+
+Once forked, clone your copy of the OpenJSCAD.org components.
+```
+git clone https://github.com/myusername/OpenJSCAD.org.git
+cd OpenJSCAD.org
+```
+
+**We suggest downloading NPM. This will allow you to generate API documents, and run test suites.**
+
+Once you have NPM, install all the dependencies for development.
+```
+npm install
+```
+And, run the test cases to verify the library is functional.
+```
+npm test
+```
+Next, start a small webserver to run the website. (Note: The web pages will automatically reload if changes are made.)
+```
+npm run start-dev
+```
+
+The project is structured as:
+- *.css : style sheets for the pages
+- index.html : main page with all features enabled
+- viewer-minimal.html : example of how to use just the veiwer
+- viewer-options.html : example of how to control the viewer options
+- dist/*.js : specific BAR files that are built fort each page
+- src/
+  - *.js : ??
+  - cli/*.js : ??
+  - core/*.js : ??
+  - io/*.js : ??
+  - jscad/*.js : ??
+  - ui/
+    - *.js : ??
+    - viewer/*.js : ??
+  - utils/*.js : various utiliity functions
+
+You can now make changes to the library, as well as the test suites.
+
+If you intend to contribute changes back to JSCAD then please be sure to create test cases.
+
+Done? Great! Push the changes back to the fork.
```

Comment:
I still think this is both too much and too little detail, let us find a good git + github guide to include :)